### PR TITLE
Add script for e2e tests where Istio is already installed.

### PR DESCRIPTION
**Pull request by ostromart**
This script has two purposes:

Add a convenient script entry point for external test frameworks (in this case k8s e2e integration tests).
Run tests against a cluster where Istio components are already installed (e.g. where Istio is a selected k8s addon).
The caller is expected to have checked out istio repo at the desired sha1 to test against.

### CODE REVIEW

File: tests/e2e_istio_preinstalled.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+[[  $1 == "all" || $1 == "bookinfo" || $1 == "mixer" || $1 == "simple" ]] || { echo "$1 is not a legal test name"; exit 1; }
+
+declare -a tests
+[[  $1 == "all" ]] && tests=("bookinfo" "mixer" "simple") || tests=($1)
+
+HUB=gcr.io/istio-testing
+
+cd ${WORKSPACE}/istio
+git reset --hard ${TAG}
```
Comment (yutongz) [inline tests/e2e_istio_preinstalled.sh]: Where do you set ${TAG} ?

Comment (ostromart) [reply tests/e2e_istio_preinstalled.sh]: It will be passed in as an env var.

Comment (ostromart) [reply tests/e2e_istio_preinstalled.sh]: The test framework checks out master at head. We have to set it to the sha we want to test inside the script.

**Comment by yutongz** [comment]
Well, you might want to let auto-merge robot retests and auto merges the PR :)

File: tests/e2e_istio_preinstalled.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+set -o pipefail
+
+[[  $1 == "all" || $1 == "bookinfo" || $1 == "mixer" || $1 == "simple" ]] || { echo "$1 is not a legal test name"; exit 1; }
+
+declare -a tests
+[[  $1 == "all" ]] && tests=("bookinfo" "mixer" "simple") || tests=($1)
+
+HUB=gcr.io/istio-testing
+
+cd ${WORKSPACE}/istio
```
Comment (krzyzacy) [inline tests/e2e_istio_preinstalled.sh]: ah, this need to be `${WORKSPACE}/github.com/istio`

**Pull request merged**